Library/Homebrew/cmd/install.rb

```diff
@@ -242,6 +242,24 @@ def run
 
           installed_casks, new_casks = casks.partition(&:installed?)
 
+          download_queue = Homebrew::DownloadQueue.new(pour: true) if Homebrew::EnvConfig.download_concurrency > 1
+          fetch_casks = Homebrew::EnvConfig.no_install_upgrade? ? new_casks : casks
+
+          if download_queue
+            fetch_casks_sentence = fetch_casks.map { |cask| Formatter.identifier(cask.full_name) }.to_sentence
+            oh1 "Fetching downloads for: #{fetch_casks_sentence}", truncate: false
+
+            fetch_casks.each do |cask|
+              Cask::Installer.new(cask, binaries: args.binaries?, verbose: args.verbose?,
+                                                   force: args.force?, skip_cask_deps: args.skip_cask_deps?,
+                                                   require_sha: args.require_sha?, reinstall: true,
+                                                   quarantine: args.quarantine?, zap: args.zap?, download_queue:)
+                             .enqueue_downloads
+            end
```

Comment:
Can/should this be combined with the similar `each` call on line 263?

Reply:
I'll take a look 👍🏻